doc/source/cluster/vms/user-guides/launching-clusters/on-premises.md

```diff
@@ -4,8 +4,76 @@
+
+(manual-cluster-launcher)=
+
+## Using Ray cluster launcher
+
+The Ray cluster launcher is part of the `ray` command line tool. It allows you to start, stop and attach to a running ray cluster using commands such as  `ray up`, `ray down` and `ray attach`. You can use pip to install it, or follow [install ray](installation) for more detailed instructions.
+
+```bash
+# install ray
+pip install "ray[default]"
+```
+
+### Start Ray with the Ray cluster launcher
+
+The provided [example-full.yaml](https://github.com/ray-project/ray/tree/eacc763c84d47c9c5b86b26a32fd62c685be84e6/python/ray/autoscaler/local/example-full.yaml) cluster config file will create a Ray cluster given a list of nodes.
+
+Note that you'll need to fill in your [head_ip](https://github.com/ray-project/ray/blob/eacc763c84d47c9c5b86b26a32fd62c685be84e6/python/ray/autoscaler/local/example-full.yaml#L20), a list of [worker_ips](https://github.com/ray-project/ray/blob/eacc763c84d47c9c5b86b26a32fd62c685be84e6/python/ray/autoscaler/local/example-full.yaml#L26), and the [ssh_user](https://github.com/ray-project/ray/blob/eacc763c84d47c9c5b86b26a32fd62c685be84e6/python/ray/autoscaler/local/example-full.yaml#L34) field in those templates
+
+
+
+Test that it works by running the following commands from your local machine:
+
+```bash
+# Download the example-full.yaml
+wget https://raw.githubusercontent.com/ray-project/ray/master/python/ray/autoscaler/local/example-full.yaml
+
+# Update the example-full.yaml to update head_ip, worker_ips, and ssh_user.
+# vi example-full.yaml
+
+# Create or update the cluster. When the command finishes, it will print
+# out the command that can be used to SSH into the cluster head node.
+ray up example-full.yaml
+
+# Get a remote screen on the head node.
+ray attach example-full.yaml
+
+# Check the cluster status and make sure you eventually see all nodes join.
+ray status example-full.yaml
+
+# ...Try running a simple Ray program...
+python -c "import ray; ray.init(); print(ray.available_resources())"
+
+# Tear down the cluster.
+ray down example-full.yaml
+```
+
+Congratulations, you have started a local Ray cluster!
+
+Check out [job submission](jobs-quickstart) next for instructions on running Ray applications on the cluster from your laptop.
+
+### Troubleshooting: Head node launches but worker nodes do not
+
+This can happen if the head node cannot SSH into the worker nodes and/or the head and worker nodes cannot connect to each other's ports.
+
+In this case, after running ``ray up example-full.yaml`` attach to the head node and check that you can SSH into the worker node using the following commands:
+
+```bash
+# Get a remote screen on the head node.
+ray attach example-full.yaml
+
+# Check that you can SSH from the head node to the worker node.
+ssh -i ~/ray_bootstrap_key.pem <worker node IP>
+```
+
+To check that ports are accessible, use a tool like ``telnet <ip> <port>``. The ports to check are:
```

Comment:
Non-blocker: This is great! I think we can probably also add a common pitfalls or something and suggest user to check the security group with their nodes to ensure the inbound port are setup correctly :)

Reply:
Good idea, thanks!